### ISSUE TITLE
explosion at 14000/500000

### ISSUE THREAD

**Issue by frolf** [carpedm20/BEGAN-pytorch#1]
Not sure exactly what occurred, but while training with your code with default parameters and tensorboard enabled, the training lost it at 14000/500000.

![beganoops](https://cloud.githubusercontent.com/assets/25496138/24635525/c3de0be6-18a1-11e7-8a7f-d8f2128012f6.png)
![beganoops2](https://cloud.githubusercontent.com/assets/25496138/24635527/c73cd218-18a1-11e7-8682-bf5f272d17f0.png)

The network then began outputting completely black images for all three _D, _D, and _D_fake.

Not much more insight from my end I am afraid.  Trained on a Titan XP in ubuntu with pytorch 0.1.10 - py27_1cu80 [cuda80] - soumith, torchvision 0.1.6, and python 2.7.13 on ubuntu 16.04.

Loss_D went from 0.0436 to 1.45,  and L_x went from 0.0436 to 1.522

**Comment by carpedm20** [comment]
I'm working on this and other issues. Autoencoder which is D trained well but G doesn't. G seems to go to the right path only at the very early steps but when k_t goes near 0, it goes wrong.

**Comment by carpedm20** [comment]
After I switched 'z_dim' from 1024 to 124, I guess explosion with default parameter is solved.

**Comment by malreddysid** [comment]
Hi! Were you able to recreate the results given in paper?

**Comment by carpedm20** [comment]
Not yet. But I will.

**Comment by malreddysid** [comment]
https://github.com/rcalland/chainer-BEGAN
Here is another implementation which is done using chainer. It might help you.

**Comment by carpedm20** [comment]
@malreddysid Yea. I saw it too but it doesn't have a learning rate schedule which, in my opinion, important to get out of mode collapsing. I guess it's hard to understand gradient flow of pytorch and I'm working on a tensorflow implementation which is better than this implementation and actually show some faces.
![image](https://cloud.githubusercontent.com/assets/3346407/24759101/5a99ccf8-1b1f-11e7-9a02-6bdba2e15cbe.png) ![image](https://cloud.githubusercontent.com/assets/3346407/24759149/7f4c16aa-1b1f-11e7-89cc-c8acba613300.png)

**Comment by ZY-WANG-0304** [comment]
Hi! Maybe you could change the L1loss() to tensor,reducemean(tensor.abs(D-X) ).
I have try by this way. The losses seems becoming similar to your tensorflow-version，but the image generated is still bad.
